aea/mail/base.py

```diff
@@ -32,6 +32,19 @@
 logger = logging.getLogger(__name__)
 
 
+class EnvelopeContext:
+    """Extra information for the handling of an envelope."""
```

Comment:
Note, in our architecture doc this context is more broadly defined. It will be used for connection matching, in the peer to peer routing system and for the skill/handler routing. So it will be a string and we need to agree on its form. Some info like the connection which is used for routing is only needed inside the agent, other info like which `context` you talk about etc is also needed outside of it.